torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -781,6 +781,7 @@ ProcessGroupNCCL::ProcessGroupNCCL(
       ValueError,
       at::cuda::getNumGPUs() != 0,
       "ProcessGroupNCCL is only supported with GPUs, no GPUs found!");
+  const auto ncclVersion = getNcclVersion();
```

Comment:
Nit: add a comment here explaining that this call needs to stay here bc it causes the setenv race